cmd/opampsupervisor/supervisor/config/config.go

```diff
@@ -5,8 +5,22 @@ package config
 
 // Supervisor is the Supervisor config file format.
 type Supervisor struct {
-	Server *OpAMPServer
-	Agent  *Agent
+	Server       *OpAMPServer
+	Agent        *Agent
+	Capabilities *Capabilities `mapstructure:"capabilities"`
+}
+
+// Capabilities is the set of capabilities that the Supervisor supports.
+// TODO: Support more capabilities
+type Capabilities struct {
+	AcceptsRemoteConfig            *bool `mapstructure:"accepts_remote_config"`
+	AcceptsOpAMPConnectionSettings *bool `mapstructure:"accepts_opamp_connection_settings"`
+	AcceptsRestartCommand          *bool `mapstructure:"accepts_restart_command"`
```

Comment:
Same here, I think we don't have restarts.

Reply:
Updated